src/components/ReportWelcomeText.js

```diff
@@ -98,7 +97,12 @@ const ReportWelcomeText = (props) => {
                                 <Tooltip text={tooltip}>
                                     <Text
                                         style={[styles.textStrong]}
-                                        onPress={() => Navigation.navigate(ROUTES.getProfileRoute(participantAccountIDs[index]))}
+                                        onPress={() => {
+                                            const accountDetails = props.personalDetails[OptionsListUtils.addSMSDomainIfPhoneNumber(displayNamesWithTooltips[index].tooltip)];
+                                            if (accountDetails && accountDetails.accountID) {
+                                                Navigation.navigate(ROUTES.getProfileRoute(accountDetails.accountID));
+                                            }
```

Comment:
Tagging @Beamanator as I believe this might have to be included in your refactor PR